kernel/indTyping.ml

```diff
@@ -406,7 +407,7 @@ let typecheck_inductive env ~sec_univs (mie:mutual_inductive_entry) =
     | TemplateArity _ -> true
     | RegularArity _ -> false
     in
-    if List.exists check data then Some (get_template univs params data) else None
+    if List.exists check data then Some (get_template mie.mind_entry_universes params data) else None
```

Comment:
In the "merge the template boolean" commit this could be simplified to `get_template ...` with get_template returning None for mono and poly. Then there are no anomalies.

Reply:
That would result in an error at some point, if the kernel is passed template then it is obligated to produce template (IIRC)

The code flow is
- template boolean determines if ind_min_univ is Some in the univ info 
- if ind_min_univ is Some then we get TemplateArity (so it's everywhere or nowhere)
- ind_min_univ is only modified by Option.map so we get TemplateArity iff we had the template boolean

Reply:
This looks a bit fragile to enforce this invariant right now. While it probably holds, I'd prefer deferring this to a later PR. Separation of concerns, as we say.